### ISSUE TITLE
Does Android count as Linux for steed?

### ISSUE THREAD

**Issue by tbu-** [japaric/steed#52]
Do we support Android?

**Comment by japaric** [comment]
> Do we support Android?

We can't list Android as a supported platform because we are not running tests for it. I mean, it *might* work because Android is based on Linux but we are not committed to fixing any problem on that platform because it's not listed as a [supported platform](https://github.com/japaric/steed#supported-architectures).

That being said, I'm :+1: on supporting Android if anyone figures out how to test it inside Travis CI or through some other testing infra.

**Comment by parched** [comment]
Yes, it should just be the tests. Static musl binaries work on both Android and normal linux so there's no reason why steed shouldn't /couldn't.

**Comment by briansmith** [comment]
My understanding is that, despite being based on Linux, Android doesn't have any stable C ABI other than bionic (Android's libc). For example, they may choose to fix a kernel bug in bionic rather than in the kernel. Or they may choose to not fix a serious kernel vulnerability at all, if it isn't triggered by bionic. Or they may have other libraries that a Rust program would link with that assume bionic in some deadly-if-not-bionic way. Therefore there's always going to be non-trivial risk in using the Kernel directly on Android, that isn't present in general in Linux.

Also, Android applications are probably more binary-size-conscious than desktop/server applications.
(I personally am taking a wait-and-see approach to see how steed would affect the binary size of Android apps.)

**Comment by anatol** [comment]
Android's kernel ABI is stable as much as underlying Linux ABI. It is safe to use `syscall` interface. It is what golang android is doing.

Saying it Android `bionic` is slightly different from `glibc`. For example DNS name resolution at android works different way than in `glibc`. Android target will need to account for it.
